src/shared/user-helpers.ts

```diff
@@ -1,34 +1,44 @@
 import { UserProfile } from "@auth0/nextjs-auth0";
+import { default as gravatarUrl, Options } from "gravatar-url";
 
 import { fetchUser, patchUser, postUser } from "../requests/users";
 import { User } from "../types";
 
-export const baseUrl = process.env.AUTH0_BASE_URL ?? "";
-
-export const userProfileToUser = (userProfile: UserProfile): User => {
-  return {
-    id: userProfile.sub,
-    name: userProfile.name,
-    nickname: userProfile.nickname,
-    email: userProfile.email,
-    updateDate: userProfile.updated_at,
-  } as User;
+const defaultGravatarUrlOptions: Options = {
+  size: 100,
+  rating: "g",
+  default: "mp",
 };
 
+export const baseUrl = process.env.AUTH0_BASE_URL ?? "";
+
 export const createOrUpdateDbUser = async (
-  userId: string,
-  userData: User,
+  userProfile: UserProfile,
 ): Promise<User | undefined> => {
+  const userId = userProfile.sub as string;
   if (!userId) {
-    return;
+    throw new Error("userProfile.sub is required");
   }
 
+  const {
+    sub: id = userId,
+    updated_at: updateDate,
+    picture,
+    ...profileData
+  } = userProfile;
+
+  const avatarUrl: string =
+    picture ?? gravatarUrl(userProfile.email ?? "", defaultGravatarUrlOptions);
+
+  const userData: User = { id, avatarUrl, updateDate, ...profileData };
```

Comment:
Here the eslint shows warning, do not successed resolving it. Giving up.

Reply:
I applied a hotfix by adding `as Omit<User, "createDate">`. We already do this in a few other places like `userProfile.sub as string`. Technically speaking, TypeScript is correct in its warnings. Auth0’s `UserProfile` allows `null` or `undefined`, while our Prisma User does not. This is a potential source of unexpected runtime errors. I have a small idea to explore, let’s do this after merging this PR and #56